R/expand.R

```diff
@@ -33,7 +30,11 @@ expand <- function(x, approx_range) UseMethod("expand")
 #' "2008-XX-31", "..2002-02-03", "2001-01-03..", "28 BC"))
 #' expand(d)
 #' @export
-expand.mdate <- function(x, approx_range = 0) {
+expand <- function(x, approx_range = 0) {
+  if (!is_messydate(x)) {
+    message("Date object(s) converted to 'mdate' class")
+    x <- as_messydate(x)
+  }
```

Comment:
Why not keep it as a method, and then add additional class functionality that first coerces to messydate? In other words, how will this respond if I try to 'expand' some numeric vector, or a list, or something else unexpected?

Reply:
Thank you for the feedback! Adding additional class functionality would require we setup "expand" for various possible date classes (e.g. date, character, POSIXct, ... ). That is why I reasoned transforming it into a function as we have for `contract()`, for example, could work. However, I understand that by doing so we might make the function "too flexible". Maybe we should go back to only working with 'mdate' class, what do you think?